pkg/virt-operator/resource/generate/components/util.go

```diff
@@ -0,0 +1,13 @@
+package components
+
+import "fmt"
+
+func GeneratePromQlQueryForValueDifferenceInHourInterval(metricsToSumUpAndCalculateDifferenceInInterval string) string {
+	return GeneratePromQlQueryForValueDifferenceInInterval(metricsToSumUpAndCalculateDifferenceInInterval, 60)
+}
+func GeneratePromQlQueryForValueDifferenceInFiveMinutesInterval(metricsToSumUpAndCalculateDifferenceInInterval string) string {
+	return GeneratePromQlQueryForValueDifferenceInInterval(metricsToSumUpAndCalculateDifferenceInInterval, 5)
+}
+func GeneratePromQlQueryForValueDifferenceInInterval(metricsToSumUpAndCalculateDifferenceInInterval string, timeIntervalInMinutes int) string {
```

Comment:
nits: IMO the function names and parameters are very long, which makes it a bit hard to read. I would suggest:
`GeneratePromQlQueryForValueDifferenceInHourInterval()` -> `GenerateValueDifferenceForLastHour()`
`GeneratePromQlQueryForValueDifferenceInFiveMinutesInterval()` -> `GenerateValueDifferenceForLastMinute()`
`GeneratePromQlQueryForValueDifferenceInInterval()` -> `GenerateValueDifferenceForInterval()`
`metricsToSumUpAndCalculateDifferenceInInterval` -> `metric`

Reply:
Also - please make all functions here private. I don't see a reason for them being public.

Reply:
"of course, another function can be made for rest calls with errors." - I'm not sure about 'code' label's value and it's source but if you consider a case where different pods have different error codes we would have to create 2 function for each kind of pod.
also i tried to stay as much generic as possible and i think that this function might be useful for other kind of metrics such as number of migration in time interval and in that case we can't be sure that migration metric would have code label at all.

Reply:
Done.
Thanks!